docs/internals/contributing/writing-documentation.txt

```diff
@@ -341,6 +341,39 @@ documentation:
     Five
     ^^^^
 
+* Add headers for functions, classes, and methods in reference documentation:
+
+  - Each function, class, method, or attribute in reference documentation should
+    have a header using the appropriate heading level based on document hierarchy.
+  - Use the exact name of the item as the header text.
```

Comment:
```suggestion
  - Use the exact name of the item as the header text. Headers for functions
    or methods should have brackets appended, e.g. ``patch_cache_control()``.
```

I think it might be helpful to explicitly declare this in the guidelines.